models/repo_language_stats.go

```diff
@@ -144,3 +144,37 @@ func (repo *Repository) UpdateLanguageStats(commitID string, stats map[string]fl
 
 	return sess.Commit()
 }
+
+// CopyLanguageStat Copy originalRepo language stat information to destRepo (use for forked repo)
+func CopyLanguageStat(originalRepo, destRepo *Repository) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	originalRepoLang := make(LanguageStatList, 0, 6)
+	destRepoLang := make(LanguageStatList, 0, 6)
+	if err := sess.Where("`repo_id` = ?", originalRepo.ID).Desc("`percentage`").Find(&originalRepoLang); err != nil {
+		return err
+	}
+	if len(originalRepoLang) > 0 {
```

Comment:
```suggestion
	destRepoLang := make(LanguageStatList, 0, len(originalRepoLang))
	if len(originalRepoLang) > 0 {
```

Reply:
explanation: xorm can resize the originalRepoLang so create destRepoLang afterwards already right sized